core/src/main/java/org/kiwix/kiwixmobile/core/error/ErrorActivity.kt

```diff
@@ -0,0 +1,233 @@
+      extras.getSerializable(EXCEPTION_KEY) as Throwable
+    } else {
+      null
+    }
+    setupReportButton()
+    restartButton.setOnClickListener { restartApp() }
+  }
+
+  private fun setupReportButton() {
+    reportButton.setOnClickListener {
+      startActivityForResult(
+        Intent.createChooser(emailIntent(), "Send email..."), 1
+      )
+    }
+  }
+
+  private fun emailIntent(): Intent {
+    return Intent(Intent.ACTION_SEND).apply {
+      type = "vnd.android.cursor.dir/email"
+      putExtra(
+        Intent.EXTRA_EMAIL,
+        arrayOf("android-crash-feedback@kiwix.org")
+      )
+      putExtra(Intent.EXTRA_SUBJECT, subject)
+      putExtra(Intent.EXTRA_TEXT, buildBody())
+      if (allowLogs.isChecked) {
+        val file = fileLogger.writeLogFile(this@ErrorActivity)
+        val path =
+          FileProvider.getUriForFile(
+            this@ErrorActivity,
+            applicationContext.packageName + ".fileprovider",
+            file
+          )
+        addFlags(android.content.Intent.FLAG_GRANT_READ_URI_PERMISSION)
+        putExtra(android.content.Intent.EXTRA_STREAM, path)
+      }
+    }
+  }
+
+  private fun buildBody(): String = """ 
+  $initialBody
+    
+  ${if (allowCrash.isChecked && exception != null) exceptionDetails() else ""}
+  ${if (allowZims.isChecked) zimFiles() else ""}
+  ${if (allowLanguage.isChecked) languageLocale() else ""}
+  ${if (allowDeviceDetails.isChecked) deviceDetails() else ""}
+  ${if (allowFileSystemDetails.isChecked) systemDetails() else ""}
+  """.trimIndent()
+
+  private fun exceptionDetails(): String =
+    """
+    Exception Details:
+    ${toStackTraceString(exception!!)}
+    """.trimIndent()
+
+  private fun zimFiles(): String {
+    val allZimFiles = bookDao.getBooks().joinToString {
+      """
+      ${it.book.getTitle()}:
+      Articles: [${it.book.getArticleCount()}]
+      Creator: [${it.book.getCreator()}]
+      """.trimIndent()
+    }
+    return """
+        Current Zim File:
+        ${zimReaderContainer.zimCanonicalPath}
+        All Zim Files in DB:
+        $allZimFiles
+        
+        """.trimIndent()
+  }
+
+  private fun languageLocale(): String = """
+    Current Locale:
+    ${getCurrentLocale(applicationContext)}
+    
+  """.trimIndent()
+
+  private fun deviceDetails(): String = """
+    BluetoothClass.Device Details:
+    Device:[${Build.DEVICE}]
+    Model:[${Build.MODEL}]
+    Manufacturer:[${Build.MANUFACTURER}]
+    Time:[${Build.TIME}]
+    Android Version:[${Build.VERSION.RELEASE}]
+    App Version:[$versionName $versionCode]
+    
+  """.trimIndent()
+
+  private fun systemDetails(): String = """
+    Mount Points
+    ${mountPointProducer.produce().joinToString {
+    "$it\n"
+  }}
+    External Directories
+    ${externalFileDetails()}
+  """.trimIndent()
+
+  private fun externalFileDetails(): String = """
+    ${ContextCompat.getExternalFilesDirs(this, null).joinToString {
+    if (it != null)
```

Comment:
the if is a bit verbose 
```
it?.path ?: "null"
```
is equivalent

Reply:
done